src/dos/drive_fat.cpp

```diff
@@ -1023,22 +1045,26 @@ bool fatDrive::FileCreate(DOS_File **file, char *name, uint16_t attributes) {
 		if(!getDirClustNum(name, &dirClust, true)) return false;
 		memset(&fileEntry, 0, sizeof(direntry));
 		memcpy(&fileEntry.entryname, &pathName[0], 11);
-		fileEntry.attrib = (uint8_t)(attributes & 0xff);
+		fileEntry.attrib  = (uint8_t)(attributes & 0xff);
+		fileEntry.modTime = DOS_GetCurPackedTime();
+		fileEntry.modDate = DOS_GetCurPackedDate();
 		addDirectoryEntry(dirClust, fileEntry);
 
 		/* Check if file exists now */
 		if(!getFileDirEntry(name, &fileEntry, &dirClust, &subEntry)) return false;
 	}
 
 	/* Empty file created, now lets open it */
-	/* TODO: check for read-only flag and requested write access */
 	*file = new fatFile(name, fileEntry.loFirstClust, fileEntry.entrysize, this);
-	(*file)->flags=OPEN_READWRITE;
+	if (fileEntry.attrib & DOS_ATTR_READ_ONLY) {
+		(*file)->flags = OPEN_READ;
+	} else {
+		(*file)->flags = OPEN_READWRITE;
+	}
```

Comment:
After this if/else, suggest a single C++ reinterpret cast over to a `fat_file` pointer and then use that in all three subsequent `((fatFile *)(*file))->` calls.